File: adaptive-arp-api-java/ARP/src/me/adaptive/arp/api/Endpoint.java
/*
 *
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     * Ferran Vila Conesa
 *                 <http://github.com/fnva>
 *                 <http://twitter.com/ferran_vila>
 *                 <mailto:ferran.vila.conesa@gmail.com>
 *
 *     * Aryslan
 *                 <http://github.com/Aryslan>
 *                 <http://twitter.com/Aryslan>
 *                 <mailto:ddbc@gft.com>
 *
 *     * Francisco Javier Martin Bueno
 *                 <https://github.com/kechis>
 *                 <mailto:kechis@gmail.com>
 *
 * =====================================================================================================================
 *
 */

package me.adaptive.arp.api;

/**
 * Created by FRMI on 12/08/2014.
 */
public class Endpoint {

    /**
     * The remote serice host (alias or IP).
     *
     * @since ARP1.0
     */
    private String host;
    /**
     * The remote service path (to be added to the host and port url).
     *
     * @since ARP1.0
     */
    private String path;
    /**
     * The remote service accessible port.
     *
     * @since ARP1.0
     */
    private int port;
    /**
     * The proxy url - if needed - to access the remote service. If IP and port are used, use the following syntax: "http://<IP>:<Port>".
     *
     * @since ARP1.0
     */
    private String proxy;
    /**
     * The remote service scheme.
     *
     * @since ARP1.0
     */
    private String Scheme;

    /**
     * Constructor used by the implementation
     *
     * @param host
     * @param path
     * @param port
     * @param proxy
     * @param scheme
     * @since ARP1.0
     */
    public Endpoint(String host, String path, int port, String proxy, String scheme) {
        this.host = host;
        this.path = path;
        this.port = port;
        this.proxy = proxy;
        Scheme = scheme;
    }

    /**
     * Returns the host
     *
     * @return host
     * @since ARP1.0
     */
    public String getHost() {
        return host;
    }

    /**
     * Set the host
     *
     * @param host
     * @since ARP1.0
     */
    public void setHost(String host) {
        this.host = host;
    }

    /**
     * Returns the path
     *
     * @return path
     * @since ARP1.0
     */
    public String getPath() {
        return path;
    }

    /**
     * Set the path
     *
     * @param path
     * @since ARP1.0
     */
    public void setPath(String path) {
        this.path = path;
    }

    /**
     * Returns the port
     *
     * @return port
     * @since ARP1.0
     */
    public int getPort() {
        return port;
    }

    /**
     * Set the port
     *
     * @param port
     * @since ARP1.0
     */
    public void setPort(int port) {
        this.port = port;
    }

    /**
     * Return the proxy
     *
     * @return proxy
     * @since ARP1.0
     */
    public String getProxy() {
        return proxy;
    }

    /**
     * Set the proxy
     *
     * @param proxy
     * @since ARP1.0
     */
    public void setProxy(String proxy) {
        this.proxy = proxy;
    }

    /**
     * Returns the scheme
     *
     * @return scheme
     * @since ARP1.0
     */
    public String getScheme() {
        return Scheme;
    }

    /**
     * Set the scheme
     *
     * @param scheme
     * @since ARP1.0
     */
    public void setScheme(String scheme) {
        Scheme = scheme;
    }
}

File: adaptive-arp-api-csharp/Adaptive.Arp.Api/Adaptive/Arp/Api/Endpoint.cs
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     *
 *
 * =====================================================================================================================
 */

using Sharpen;

namespace Adaptive.Arp.Api
{
	/// <summary>Structure representing a remote or local service access end-point.</summary>
	/// <remarks>Structure representing a remote or local service access end-point.</remarks>
	public class Endpoint
	{
		/// <summary>The remote serice host (alias or IP).</summary>
		/// <remarks>The remote serice host (alias or IP).</remarks>
		/// <since>ARP1.0</since>
		private string host;

		/// <summary>The remote service path (to be added to the host and port url).</summary>
		/// <remarks>The remote service path (to be added to the host and port url).</remarks>
		/// <since>ARP1.0</since>
		private string path;

		/// <summary>The remote service accessible port.</summary>
		/// <remarks>The remote service accessible port.</remarks>
		/// <since>ARP1.0</since>
		private int port;

		/// <summary>The proxy url - if needed - to access the remote service.</summary>
		/// <remarks>The proxy url - if needed - to access the remote service. If IP and port are used, use the following syntax: "http://<IP>:<Port>".
		/// 	</remarks>
		/// <since>ARP1.0</since>
		private string proxy;

		/// <summary>The remote service scheme.</summary>
		/// <remarks>The remote service scheme.</remarks>
		/// <since>ARP1.0</since>
		private string Scheme;

		/// <summary>Constructor used by the implementation</summary>
		/// <param name="host"></param>
		/// <param name="path"></param>
		/// <param name="port"></param>
		/// <param name="proxy"></param>
		/// <param name="scheme"></param>
		/// <since>ARP1.0</since>
		public Endpoint(string host, string path, int port, string proxy, string scheme)
		{
			this.host = host;
			this.path = path;
			this.port = port;
			this.proxy = proxy;
			Scheme = scheme;
		}

		/// <summary>Returns the host</summary>
		/// <returns>host</returns>
		/// <since>ARP1.0</since>
		public virtual string GetHost()
		{
			return host;
		}

		/// <summary>Set the host</summary>
		/// <param name="host"></param>
		/// <since>ARP1.0</since>
		public virtual void SetHost(string host)
		{
			this.host = host;
		}

		/// <summary>Returns the path</summary>
		/// <returns>path</returns>
		/// <since>ARP1.0</since>
		public virtual string GetPath()
		{
			return path;
		}

		/// <summary>Set the path</summary>
		/// <param name="path"></param>
		/// <since>ARP1.0</since>
		public virtual void SetPath(string path)
		{
			this.path = path;
		}

		/// <summary>Returns the port</summary>
		/// <returns>port</returns>
		/// <since>ARP1.0</since>
		public virtual int GetPort()
		{
			return port;
		}

		/// <summary>Set the port</summary>
		/// <param name="port"></param>
		/// <since>ARP1.0</since>
		public virtual void SetPort(int port)
		{
			this.port = port;
		}

		/// <summary>Return the proxy</summary>
		/// <returns>proxy</returns>
		/// <since>ARP1.0</since>
		public virtual string GetProxy()
		{
			return proxy;
		}

		/// <summary>Set the proxy</summary>
		/// <param name="proxy"></param>
		/// <since>ARP1.0</since>
		public virtual void SetProxy(string proxy)
		{
			this.proxy = proxy;
		}

		/// <summary>Returns the scheme</summary>
		/// <returns>scheme</returns>
		/// <since>ARP1.0</since>
		public virtual string GetScheme()
		{
			return Scheme;
		}

		/// <summary>Set the scheme</summary>
		/// <param name="scheme"></param>
		/// <since>ARP1.0</since>
		public virtual void SetScheme(string scheme)
		{
			Scheme = scheme;
		}
	}
}
